teammapper-frontend/src/app/core/services/mmp/mmp.service.ts

```diff
@@ -314,8 +319,27 @@ export class MmpService implements OnDestroy {
    * Copy a node with his children in the mmp clipboard.
    * If id is not specified, copy the selected node.
    */
-  public copyNode(nodeId?: string) {
-    this.currentMap.instance.copyNode(nodeId);
+  public async copyNode(nodeId?: string) {
+    try {
+      this.currentMap.instance.copyNode(nodeId);
+
+      const successMessage = await this.utilsService.translate(
+        'TOASTS.NODE_COPIED'
+      );
+      this.toastrService.success(successMessage);
+    } catch (e) {
+      if (e.message == 'The root node can not be copied') {
```

Comment:
do we have to check for the message or isn't there a specific class / type field available?

Reply:
if not, maybe its a good idea to at least refactor this message as a constant variable. Better in the mmp lib, otherwise here at the top of the file

Reply:
Don't think this has a specific type, but strict equality didn't work for some reason (even though e.message should be typeof string, but maybe it's JS doing JS things), was my first preference as well.

Question is if we've got any other errors this could potentially throw - right now I'm just seeing a specific error message when trying to copy the root node, otherwise it doesn't actually error out at all. If we had multiple errors I could see the value in a constant, but not sure if it's worth it for a one-off.